src/plugins/data/server/field_formats/field_formats_service.ts

```diff
@@ -0,0 +1,59 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { has } from 'lodash';
+import { fieldFormats } from '../../common/field_formats';
+import { IUiSettingsClient } from '../../../../core/server';
+
+export class FieldFormatsService {
+  private readonly fieldFormatClasses: fieldFormats.IFieldFormatType[] =
+    fieldFormats.baseFormatters;
+
+  public setup() {
+    return {
+      register: (customFieldFormat: fieldFormats.IFieldFormatType) =>
+        this.fieldFormatClasses.push(customFieldFormat),
+    };
+  }
+
+  public start() {
+    return {
+      fieldFormatServiceFactory: async (uiSettings: IUiSettingsClient) => {
```

Comment:
Isn't this exactly the same as returning the content of the registry in the public plugin?